ext/fg/js/frontend.js

```diff
@@ -26,24 +26,23 @@
  */
 
 class Frontend extends TextScanner {
-    constructor(popup) {
+    constructor(popup, url) {
```

Comment:
This, or some other new class, will need to handle URL changes for the root frame.

https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event

Reply:
It seems that it's not possible to detect a `history.pushState` call without monkey patching the object. It could work similar to `FrameOffsetForwarder` asking the parent `Frontend` for the root frame or iframe URL since the context URL is for the current frame.

Reply:
Actually that's not possible here, but the popup proxy framework could be used instead.

Reply:
Oh right, I forgot about that. I'm not sure if you can even monkey patch it due to the script context isolation. Not sure what the best workaround would be in that case.

Reply:
https://github.com/FooSoft/yomichan/pull/464/commits/b050333078b12f2ae14f49a46c04085b8e11d0b8 and https://github.com/FooSoft/yomichan/pull/464/commits/2b2445ec31f0a1d05f1cabf30caeefb558195aa7